skfs/src/FixedDir.sk

```diff
@@ -111,26 +122,155 @@ class FixedDir<T: frozen> private {
       keyComparison match {
       | LT() -> continue
       | GT() -> break false
-      | EQ() -> data[i - 1].value.i0.compare(data[i].value.i0)
+      | EQ() -> data[i - 1].source.compare(data[i].source)
       } match {
       | LT() -> continue
       | EQ() -> continue
       | GT() -> break false
       };
     };
     if (!sorted) {
-      data.sortBy(row ~> (row.key, row.value.i0));
+      data.sortBy(row ~> (row.key, row.source));
     };
     _ = static::computeTags(data, 0, data.size() - 1);
-    FixedDir{data => data.toArray()}
+    static{data => data.toArray()}
+  }
+
+  protected fun decode(row: FixedRow<T>): FixedRow<T> {
+    row
+  }
+}
+
+value class CompactRow<T>(
+  values: Array<T>,
+  repeat: Int,
```

Comment:
[for another diff?] In practice it's very rare for this to be anything other than 1, I believe. It can happen that this is larger than 1 if values are inserted atomically and replication sends them across - this is the only case I'm aware of. I'm not aware of anywhere where this can be < 1. In fact there might even be an invariant somewhere.

Then from a user perspective, when are you ever inserting duplicate rows in a transaction? I honestly can't manufacture even the most artificial scenario.

Given this, might make sense to assume a repeat of 1, save ourselves 4 bytes per row, and just pay an up front check that there are no repeats > 1. Or we might even just completely remove repeats. Replication can send a count still but we could expand this using different sources (possibly, would need to check logic here carefully).